src/item.cpp

```diff
@@ -2354,76 +2271,252 @@ std::string item::tname( unsigned int quantity, bool with_prefix ) const
+            } else {
+                damtext = string_format( "%s ", get_base_material().dmg_adj( damage() ) );
+            }
+        }
+        
+        if( !faults.empty() ) {
+            damtext.insert( 0, _( "faulty " ) );
+        }
+
+        if( with_prefix && !made_of( LIQUID ) ) {
+            if( volume() >= 1000_ml && burnt * 125_ml >= volume() ) {
+                burntext = pgettext( "burnt adjective", "badly burnt " );
+            } else if( burnt > 0 ) {
+                burntext = pgettext( "burnt adjective", "burnt " );
+            }
+        }
+    }
+
+    std::string vehtext = "";
+    if( is_engine() && engine_displacement() > 0 ) {
+        vehtext = string_format( pgettext( "vehicle adjective", "%2.1fL " ), engine_displacement() / 100.0f );
+
+    } else if( is_wheel() && type->wheel->diameter > 0 ) {
+        vehtext = string_format( pgettext( "vehicle adjective", "%d\" " ), type->wheel->diameter );
+    }
+
+    std::string maintext = tname_generate_maintext( quantity );
+
+    std::string tagtext = tname_generate_tagtext();
 
     std::string modtext;
     if( gunmod_find( "barrel_small" ) ) {
         modtext += _( "sawn-off " );
     }
+    
     if( has_flag( "DIAMOND" ) ) {
         modtext += std::string( _( "diamond" ) ) + " ";
     }
 
-    ret.str( "" );
     //~ This is a string to construct the item name as it is displayed. This format string has been added for maximum flexibility. The strings are: %1$s: Damage text (e.g. "bruised"). %2$s: burn adjectives (e.g. "burnt"). %3$s: tool modifier text (e.g. "atomic"). %4$s: vehicle part text (e.g. "3.8-Liter"). $5$s: main item text (e.g. "apple"). %6s: tags (e.g. "(wet) (fits)").
-    ret << string_format( _( "%1$s%2$s%3$s%4$s%5$s%6$s" ), damtext.c_str(), burntext.c_str(),
-                          modtext.c_str(), vehtext.c_str(), maintext.c_str(), tagtext.c_str() );
+    std::string final = string_format( _( "%1$s%2$s%3$s%4$s%5$s%6$s" ), damtext,
+                                    burntext, modtext, vehtext, maintext, tagtext );
 
     if( item_vars.find( "item_note" ) != item_vars.end() ) {
         //~ %s is an item name. This style is used to denote items with notes.
-        return string_format( _( "*%s*" ), ret.str().c_str() );
+        return string_format( _( "*%s*" ), final );
+    }
+
+    return final;
+}
+
+enum naming_style : int {
+    NAMING_STANDARD,            // Wrench
+    NAMING_WITH_CHARGES,        // 00 Shot (10), Flashlight (100)
+    NAMING_WITH_AMMO_TYPE,      // Glock 18 (15) (9x19mm)
+    NAMING_CONTAINER_LEADING,   // plastic bottle with clean water (2)
+    NAMING_CONTAINER_TRAILING   // clean water (2) in plastic bottle
+};
+
+naming_style item::get_naming_style() const
+{
+    if ( is_container() && !contents.empty() ) {
+        // Non-empty container-type items
+        return NAMING_CONTAINER_TRAILING;
+    } else if ( is_gun() || is_magazine() ) {
+        // Guns and magazines
```

Comment:
Yes, that is what the if condition above says. Why are you repeating the code as comment?

Btw. it would be good if you use apply our [coding style](../../tree/master/doc/CODE_STYLE.md) - specifically: no space before `(`.

Reply:
Because many of the conditions are less obvious and need comments. In that situation I think it's appropriate to comment all of the conditions including the more obvious ones. If you don't want comments in the code I can remove them.

Also, that 'coding style' link leads to a page that simply says Not Found".

Reply:
> Because many of the conditions are less obvious and need comments.

Then comment them as you already do. If the code needs an explanation, then add it.

> In that situation I think it's appropriate to comment all of the conditions including the more obvious ones.

I don't get this. For symmetry? For consistency? To make it look nice?

> If you don't want comments in the code I can remove them.

Redundant comments like this (literally repeating the code) add no information. Reading and understanding them is a waste of time. They are neither needed nor useful. So why do they exist?

Really good code don't even need comments because the code tells the reader everything they need to know. Comments are there to help the reader understand why something is done in a certain (non-obvious) way.

--

> Also, that 'coding style' link leads to a page that simply says "Not Found".

The beauty of relative links. It does work when viewed from the https://github.com/CleverRaven/Cataclysm-DDA/pull/23722/files - here is the absolute link: https://github.com/CleverRaven/Cataclysm-DDA/tree/master/doc/CODE_STYLE.md - you also find this file in your local checkout of the project as "doc/CODE_STYLE.md".